Python/errors.c

```diff
@@ -510,6 +510,30 @@ _PyErr_GetExcInfo(PyThreadState *tstate,
     Py_XINCREF(*p_traceback);
 }
 
+PyObject*
+_PyErr_GetActiveException(PyThreadState *tstate)
+{
+    _PyErr_StackItem *exc_info = _PyErr_GetTopmostException(tstate);
+    PyObject *exc = exc_info->exc_value;
+    Py_XINCREF(exc);
+    return exc;
+}
+
+PyObject*
+PyErr_GetActiveException()
+{
+    PyThreadState *tstate = _PyThreadState_GET();
+    return _PyErr_GetActiveException(tstate);
+}
+
+void
+PyErr_SetActiveException(PyObject *exc)
+{
+    PyThreadState *tstate = _PyThreadState_GET();
+    PyObject *oldexc = tstate->exc_info->exc_value;
+    tstate->exc_info->exc_value = exc;
```

Comment:
New C API functions should not steal references: https://devguide.python.org/c-api/#public-c-api

```suggestion
    tstate->exc_info->exc_value = Py_XNewRef(exc);
```